profile/src/main/java/com/scottlogic/deg/profile/reader/AtomicConstraintTypeReaderMap.java

```diff
@@ -165,10 +167,33 @@ public boolean isDelayedConstraintsEnabled() {
         if (isDelayedConstraintsEnabled) {
 
             map.put(IS_EQUAL_TO_FIELD,
-                (dto, fields) -> new IsEqualToDynamicDateConstraint(
-                    new EqualToConstraint(fields.getByName(dto.field), getValidatedValue(dto)),
-                    fields.getByName(ConstraintReaderHelpers.getValueAsString(dto))
-                ));
+                (dto, fields) -> {
+                    if (dto.offset != null && dto.offsetUnit != null) {
+                        ChronoUnitWorkingDayWrapper unit;
+                        if (((String) dto.offsetUnit).toUpperCase().equals("WORKING DAYS")) {
+                            unit = new ChronoUnitWorkingDayWrapper(
+                                ChronoUnit.valueOf(ChronoUnit.class, ("DAYS").toUpperCase()),
+                                    true);
+                        } else {
+                            unit = new ChronoUnitWorkingDayWrapper(
+                                ChronoUnit.valueOf(ChronoUnit.class, ((String) dto.offsetUnit).toUpperCase()),
+                                false);
+                        }
+                        int value = Integer.parseInt((String) dto.offset);
+
+                        return new IsEqualToDynamicDateConstraint(
+                            new EqualToConstraint(fields.getByName(dto.field), getValidatedValue(dto)),
+                            fields.getByName(getValueAsString(dto)),
+                            unit,
+                            value
+                        );
+                    }
+
+                    return new IsEqualToDynamicDateConstraint(
+                        new EqualToConstraint(fields.getByName(dto.field), getValidatedValue(dto)),
+                        fields.getByName(getValueAsString(dto))
+                    );
+                });
```

Comment:
Might be good to have a few tests for this code now that there has been so much added.